src/main/java/uk/gov/ons/ctp/common/message/rabbit/SimpleMessageBase.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.ons.ctp.common.message.rabbit;
+
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitAdmin;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+
+public class SimpleMessageBase {
```

Comment:
The class design in this PR feels weird - the logical relationship between SMBase, SMListener and SMSender doesn't feel very strong, especially the relationship between SMListener and SMSender. It almost feels like the 3 should be rolled up into one class; I can't see where the polymorphism offered by this structure would be of much use.

Maybe there's something I'm missing!

Reply:
I'd agree

Reply:
Hmm tests seem to be net consumers or net producers of messages and rarely do both.  To my mind the contract of each is clearer when looking at the clients with them separate but I am happy to park this PR until we have more time to discuss it.

Reply:
I realise the use of inheritance here wasn't polymorphic, but inheritance isn't always polymorphic and vice versa.   Sure, if I'd created a bunch of interfaces but this is a base class containing some common code. I could switch it to composing SimpleMessageBase instead of inheriting but that would be more code and, to my eyes, more complexity.

Reply:
LOL I was hoping to be looking at golang this morning but I find myself thinking about this instead.  I did it like that because instinct told me to - it's interesting to attempt to unwind the thought process behind it.  Inheritance forms a much tighter coupling than composition which, 7.5 times out of 10 is a bad thing.  However in this case, the base class is tightly coupled to the Spring AMQP framework as are the derived classes.  Inheritance adds no additional coupling in real terms and serves to highlight the coupling.  It would not be possible to replace any part of this without replacing it all, something that would be less clear if SimpleMessageBase was composed instead of inherited.

Reply:
Sure. Personally I have a preference to composing here but I'm not very au fait with Spring AMQP so am happy to let this go :)